pandas/tests/series/indexing/test_getitem.py

```diff
@@ -234,10 +234,12 @@ def test_getitem_partial_str_slice_high_reso_with_timedeltaindex(self):
         result = ser["1 days, 10:11:12.001001"]
         assert result == ser.iloc[1001]
 
-    def test_getitem_slice_2d(self, datetime_series):
+    def test_getitem_slice_2d(self, datetime_series, using_array_manager):
         # GH#30588 multi-dimensional indexing deprecated
 
-        with tm.assert_produces_warning(FutureWarning):
+        with tm.assert_produces_warning(
+            FutureWarning, check_stacklevel=not using_array_manager
+        ):
```

Comment:
That doesn't seem to solve the wrong stacklevel. 

Now, since the ArrayManager would only be widely used in a next major version of pandas, at which point this deprecation warning is already removed, I wouldn't care too much about the correct stacklevel in this case.